druid/src/widget/split.rs

```diff
@@ -30,6 +30,15 @@ pub struct Split<T> {
     min_bar_area: f64,    // Integers only
     solid: bool,
     draggable: bool,
+    /// The split bar is hovered by the mouse. This state is locked to `true` if the
+    /// widget is active (the bar is being dragged) to avoid cursor and painting jitter
+    /// if the mouse moves faster than the layout and temporarily gets outside of the
+    /// bar area while still being dragged.
+    is_bar_hover: bool,
+    /// Offset from the split point (bar center) to the actual mouse position when the
+    /// bar was clicked. This is used to ensure a click without mouse move is a no-op,
+    /// instead of re-centering the bar on the mouse.
+    click_offset: f64,
```

Comment:
@cmyr I actually found and fixed some other behavior while refactoring as explained in the other comment : when clicking and releasing without moving the mouse, because the split point is conceptually 1 unit but the bar is generally larger, we were snapping the bar to the actual mouse cursor position. I checked in other implementations like VSCode and they don't do that; instead they consider that no mouse move is a no-op. To do that, I added a delta between the click position and the actual split point. That ensures that the bar is always moved such that the mouse cursor appears to be static (in the split direction) relative to the bar.